### PR TITLE
[CodeQuality] Register MoveOutMethodCallInsideIfConditionRector to config code-quality set

### PR DESCRIPTION
**Pull request by samsonasik**

### CODE REVIEW

**Comment by samsonasik** [comment]
@TomasVotruba PHPUnit error seems due latest PHPStan release https://github.com/phpstan/phpstan/releases/tag/0.12.51

File: packages/post-rector/src/NodeAnalyzer/NetteInjectDetector.php [hunk 1]
```diff
@@ -45,8 +45,9 @@ private function isInjectPropertyAlreadyInTheClass(Class_ $class): bool
             if ($phpDocInfo === null) {
                 continue;
             }
+            $phpDocInfoGetTagsByName = $phpDocInfo->getTagsByName('inject');
 
-            if ($phpDocInfo->getTagsByName('inject') === []) {
+            if ($phpDocInfoGetTagsByName === []) {
```
Comment (samsonasik) [inline packages/post-rector/src/NodeAnalyzer/NetteInjectDetector.php:50]: It seems not handled by `$this->expectedNameResolver->resolveForCall()`, need a new tweak

Comment (samsonasik) [reply packages/post-rector/src/NodeAnalyzer/NetteInjectDetector.php:50]: @TomasVotruba How can this be changing to : 

```php
$phpDocTagNode = $phpDocInfo->getTagsByName('inject');

# or better
$injectPhpDocTagNode = $phpDocInfo->getTagsByName('inject');
```

as you requested at https://github.com/rectorphp/rector/pull/4457#discussion_r508757259 ? The nearest probably change to:

```php
$injectphpDocInfoTagsName = $phpDocInfo->getTagsByName('inject');
```

By doing something like:

```php
            $get = str_ireplace('get', '', $arg0->value . $methodCallVarName . ucfirst($methodCallName));
            $by = str_ireplace('by', '', $get);

            return $by;
```

any suggestion? The `$this->expectedNameResolver->resolveForCall($methodCall)` returns null for that.

Comment (samsonasik) [reply packages/post-rector/src/NodeAnalyzer/NetteInjectDetector.php]: @TomasVotruba I've updated to be renamed to: 

```diff
-            if ($phpDocInfo->getTagsByName('inject') === []) {
+            if ($injectphpDocInfoTagsName === []) {
```

As my commented above for nearest possible name.

**Comment by samsonasik** [comment]
I've updated config service that require autowire, autoconfigure, public as well.

**Comment by samsonasik** [comment]
@TomasVotruba there is still possible bug, eg came from parameter, eg:

```diff
public function foo($variableobj)
{
+    $variableobj = $obj->get('variable');
+    if ($variableobj) {
-    if ($obj->get('variable')) {

      }
}
```

which `variableobj` from parameter replaced.

**Comment by samsonasik** [comment]
rebased, add fixture failing tests for variable exists from param or previous statement.

**Comment by samsonasik** [comment]
I've added handling variable exists in previous statement and/or in ClassMethod/Functions parameter

**Comment by samsonasik** [comment]
close re-open to restart ci

**Comment by samsonasik** [comment]
@TomasVotruba somehow `[ci-review] Rector Rectify` not start immediately (hanging), need close-re-open to start it

**Comment by samsonasik** [comment]
Finally, All checks have passed :tada:tada:tada: @TomasVotruba it is ready for review.

**Comment by samsonasik** [comment]
rebased.

File: packages/better-php-doc-parser/src/Printer/MultilineSpaceFormatPreserver.php [hunk 2]
```diff
@@ -54,7 +54,8 @@ public function resolveCurrentPhpDocNodeText(Node $node): ?string
     public function fixMultilineDescriptions(
         AttributeAwareNodeInterface $attributeAwareNode
     ): AttributeAwareNodeInterface {
-        if (! $attributeAwareNode->getAttribute(Attribute::ORIGINAL_CONTENT)) {
+        $attributeAwareNodeContent = $attributeAwareNode->getAttribute(Attribute::ORIGINAL_CONTENT);
```
Comment (TomasVotruba) [inline packages/better-php-doc-parser/src/Printer/MultilineSpaceFormatPreserver.php:57]:
```suggestion
        $attributeAwareNodeOriginalContent = $attributeAwareNode->getAttribute(Attribute::ORIGINAL_CONTENT);
```

The value should be same as constant name `::<...>`

File: packages/post-rector/src/Rector/NameImportingPostRector.php [hunk 3]
```diff
@@ -49,7 +49,8 @@ public function __construct(
 
     public function enterNode(Node $node): ?Node
     {
-        if (! $this->parameterProvider->provideParameter(Option::AUTO_IMPORT_NAMES)) {
+        $parameterProviderNames = $this->parameterProvider->provideParameter(Option::AUTO_IMPORT_NAMES);
```
Comment (TomasVotruba) [inline packages/post-rector/src/Rector/NameImportingPostRector.php:52]:
```suggestion
        $autoImportNames = $this->parameterProvider->provideParameter(Option::AUTO_IMPORT_NAMES);
```

Only the constant should be used probably. What do you think based on other values?

File: rules/code-quality/src/Rector/Foreach_/SimplifyForeachToCoalescingRector.php [hunk 4]
```diff
@@ -152,10 +152,11 @@ private function processForeachNodeWithReturnInside(Foreach_ $foreach, Return_ $
         } else {
             return null;
         }
+        $foreachNode = $foreach->getAttribute(AttributeKey::NEXT_NODE);
```
Comment (TomasVotruba) [inline rules/code-quality/src/Rector/Foreach_/SimplifyForeachToCoalescingRector.php:155]: Same princip as above:

```suggestion
        $nextNode = $foreach->getAttribute(AttributeKey::NEXT_NODE);
```

File: rules/code-quality/tests/Rector/If_/MoveOutMethodCallInsideIfConditionRector/Fixture/if_condition_method_call_with_arg_string_has_numeric_in_first_char.php.inc [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+<?php
+
+namespace Rector\CodeQuality\Tests\Rector\If_\MoveOutMethodCallInsideIfConditionRector\Fixture;
+
+class IfConditionMethodCallWithArgHasNumericStringInFirstArg
+{
+    public function run()
+    {
+        $obj = new self();
+        $objCondition = $obj->condition('1a');
```
Comment (TomasVotruba) [inline rules/code-quality/tests/Rector/If_/MoveOutMethodCallInsideIfConditionRector/Fixture/if_condition_method_call_with_arg_string_has_numeric_in_first_char.php.inc:32]: :+1:

File: rules/coding-style/src/Node/NameImporter.php [hunk 6]
```diff
@@ -100,7 +100,8 @@ public function importName(Name $name): ?Name
 
     private function shouldSkipName(Name $name): bool
     {
-        if ($name->getAttribute(AttributeKey::VIRTUAL_NODE)) {
+        $nameNode = $name->getAttribute(AttributeKey::VIRTUAL_NODE);
+        if ($nameNode) {
```
Comment (TomasVotruba) [inline rules/coding-style/src/Node/NameImporter.php:104]:
```suggestion
        $virtualNode = $name->getAttribute(AttributeKey::VIRTUAL_NODE);
        if ($virtualNode) {
```

**Comment by TomasVotruba** [comment]
I think I've found only 1 more case. The `constant::NAME` should return `$name`.

File: packages/better-php-doc-parser/src/Printer/MultilineSpaceFormatPreserver.php (hunk 2 above)
Comment (samsonasik) [reply packages/better-php-doc-parser/src/Printer/MultilineSpaceFormatPreserver.php]: implemented.

File: packages/post-rector/src/Rector/NameImportingPostRector.php (hunk 3 above)
Comment (samsonasik) [reply packages/post-rector/src/Rector/NameImportingPostRector.php]: I implemented to use constant value.

File: rules/code-quality/src/Rector/Foreach_/SimplifyForeachToCoalescingRector.php (hunk 4 above)
Comment (samsonasik) [reply rules/code-quality/src/Rector/Foreach_/SimplifyForeachToCoalescingRector.php]: implemented.

File: rules/coding-style/src/Node/NameImporter.php (hunk 6 above)
Comment (samsonasik) [reply rules/coding-style/src/Node/NameImporter.php]: implemented.

**Comment by samsonasik** [comment]
All checks have passed :tada: All incorporated :tada: @TomasVotruba I think it is ready.

File: rules/code-quality/src/Rector/If_/MoveOutMethodCallInsideIfConditionRector.php [hunk 7]
```diff
@@ -19,12 +24,25 @@
...
+     * @var string
+     * @see https://regex101.com/r/LTykey/1
+     */
+    private const START_ALPHA_REGEX = '#^[a-zA-Z]#';
+
+    /**
+     * @var string
+     * @see https://regex101.com/r/sYIKpj/1
+     */
+    private const CONTANT_REGEX = '#(_)([a-z])#';
```
Comment (TomasVotruba) [inline rules/code-quality/src/Rector/If_/MoveOutMethodCallInsideIfConditionRector.php:44]:
```suggestion
    private const CONSTANT_REGEX = '#(_)([a-z])#';
```

File: rules/code-quality/src/Rector/If_/MoveOutMethodCallInsideIfConditionRector.php [hunk 8]
```diff
@@ -172,16 +193,79 @@ private function getVariableName(MethodCall $methodCall): ?string
...
             return $variableName;
         }
 
         $arg0 = $methodCall->args[0]->value;
         if ($arg0 instanceof ClassConstFetch && $arg0->name instanceof Identifier) {
-            $explodeUnderscore = explode('_', $arg0->name->toString());
-            return $methodCallVarName . ucfirst(strtolower((string) end($explodeUnderscore)));
+            return Strings::replace(
+                strtolower($arg0->name->toString()),
+                self::CONTANT_REGEX,
```
Comment (TomasVotruba) [inline rules/code-quality/src/Rector/If_/MoveOutMethodCallInsideIfConditionRector.php:204]:
```suggestion
                self::CONSTANT_REGEX,
```

**Comment by TomasVotruba** [comment]
Looks good enough, thank you :+1:

**Review by TomasVotruba — APPROVED** [review]

**Pull request merged**